### PR TITLE
Fixed NodeID and PieceID marshaling from JSON

### PR DESCRIPTION
**Pull request by 3bl3gamer**
What: 

Decoding fix and tests for `UnmarshalJSON` of `NodeID` and `PieceID`. Tests also added for `MarshalJSON`.

Why:

`UnmarshalJSON` expected buffer to contain data suitable for `*FromString` function but there was actually comma-surrounded JSON string (i.e expected `1Q2345`, got `"1Q2345"`).

## Code Review Checklist (to be filled out by reviewer)
 - [ ] Does the PR describe what changes are being made?
 - [ ] Does the PR describe why the changes are being made?
 - [ ] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [ ] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [ ] Could the PR be broken into smaller PRs?
 - [ ] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [ ] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [ ] Does any documentation need updating?

### CODE REVIEW

File: pkg/storj/node_test.go [hunk 1]
```diff
@@ -140,6 +141,24 @@ func TestNodeID_String_Version(t *testing.T) {
 	}
 }
 
+func TestNodeID_MarshalJSON(t *testing.T) {
+	nodeID, _ := storj.NodeIDFromString("12vha9oTFnerxYRgeQ2BZqoFrLrnmmf5UWTCY2jA77dF3YvWew7")
+	buf, err := json.Marshal(nodeID)
+	if !assert.NoError(t, err) {
+		t.Fatal()
+	}
```
Comment (kaloyan-raev) [inline pkg/storj/node_test.go]: You can simplify this check by using `require.NoError(t, err)`. It will fail the test immediately if there is an error.

**Comment by stefanbenten** [comment]
@3bl3gamer Good news, you were just being added to our CLA List 👍

**Comment by 3bl3gamer** [comment]
Thank you!

Updated tests, replaced `!assert.NoError` with `require.NoError`.

**Comment by kaloyan-raev** [comment]
@3bl3gamer I think you missed the one I made the comment on.

**Comment by 3bl3gamer** [comment]
@kaloyan-raev Oops, sorry! Fixed that.

**Comment by kaloyan-raev** [comment]
@3bl3gamer Thank you for your contribution!

**Pull request merged**